test/end-to-end/getRedirectionURL.test.js

```diff
@@ -377,5 +377,243 @@ describe("getRedirectionURL Tests", function () {
                 assert.equal(newUserCheck, "thirdpartypasswordless-true");
             });
         });
+
+        describe("No Redirection", function () {
```

Comment:
another case to test is that if we are showing the login UI in a popup AND fullpage, then you want to conditionally return `null` from `getRedirectionUrl` function. One way to do this is to use userContext when embedding the SignInUp component, and that same userContext should be available in `getRedirectionUrl`

Reply:
@anku255 and I discussed, and we concluded that we don't need to do `doRedirection` because:
- User does not actually need to pass `shouldRedirectOnSessionExists`
- The requirement to pass userContext to the `SignInUp` component is only if the dev is having a popup AND a full page auth screen - which we feel is a niche use case. In regular just popup case, you don't need to do the userContext thing.